### PR TITLE
shell completion: fix problems with container path completion

### PR DESCRIPTION
**Pull request by Luap99**
When you try to complete a path which exists and it is a file the
completion logic did not check the parent dir for other matching file
names. To fix that we have to check if the current completion is not a
dir and use the parent dir in this case.

See the updated test for an example why this is required.

Also make sure directories are correctly completed, the shell always
adds the "/" as suffix to signal the user that this path is a directory.
In this case we do not want to automatically add a space. When the path
is a regular file we want the space after the suggestion since there is
nothing more to complete.

This better matches the normal default shell completion.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by edsantiago** [comment]
hmmmm... at first glance, it looks like ubuntu sorts filenames differently. Maybe a difference in locale??

File: test/system/600-completion.bats [hunk 1]
```diff
@@ -354,23 +354,29 @@ function _check_no_suggestions() {
...
         # It is important the we will still use the image root and not escape to the host
         run_completion $cmd $IMAGE "../../"
-        assert "$output" =~ ".*^../../etc\$.*^../../home\$.*" "relative root directories suggested (cmd: podman $cmd ../../)"
+        assert "$output" =~ ".*^../../etc/\$.*^../../home/\$.*" "relative root directories suggested (cmd: podman $cmd ../../)"
     done
 
     random_name=$(random_string 30)
     random_file=$(random_string 30)
-    run_podman run --name $random_name $IMAGE touch /tmp/$random_file
+    run_podman run --name $random_name $IMAGE sh -c "touch /tmp/$random_file && touch /tmp/${random_file}2 && mkdir /emptydir"
```
Comment (edsantiago) [inline test/system/600-completion.bats:362]: Simplest (safest, maybe) suggestion: change the first to `${random_file}1` (and following checks accordingly). Some locales sort `foo2` before `foo`, but there is no locale (that I know of) that will sort `foo2` before `foo1`

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

**Comment by edsantiago** [comment]
LGTM but the lack of ordering bothers me. In particular, I'm confused because I've been using this completion feature since you introduced it, and what it suggests is reliably & consistently sorted. So much so, that I strongly believe we _can_ rely on sorting, and I'll go as far as to say that we _should_ rely on it. Maybe my f35 kernel is doing it in its `readdir`. I don't know.

I think we should guarantee sorted order. As an end user, I would not even think to assume otherwise: when I hit tab, I want to see elements in order. I don't know if that has to be solved by `sort` within podman, or by finding out what guarantees we get from Go readdir, but I think we should solve that and solve it early.

**Pull request merged**